### DISCUSSION TITLE
The Chunkloader Upgrade

### DISCUSSION THREAD

**Discussion by NeutrinoKillerino** [fnuecke/oc2#100] [Ideas]
Hello everyone, I was wondering if there are plans to bring the [Chunkloader Upgrade](https://ocdoc.cil.li/item:chunkloader_upgrade) to oc2. If not, I would like to give it a shot at adding it (it's been a few years since I've worked on Minecraft mods). Would you say it is feasible to implement this feature in the current state of the mod? The implementation would be inspired by the [Chunkloader mod](https://github.com/SuperMartijn642/ChunkLoaders).
Any pointers are welcome.

**Comment by fnuecke** [comment]
Hey! Sure thing, a robot module would definitely not be the worst place to start! Should be feasible with the current state of the mod. Would gladly have someone take something like this on! Would also give me an idea, if the current architecture is understandable for people :)

One starting point for reference would probably be the [redstone card](https://github.com/fnuecke/oc2/blob/1.18-forge/src/main/java/li/cil/oc2/common/bus/device/item/RedstoneInterfaceCardItemDevice.java). Then just follow the usages. It probably does a bit more than what you need, e.g. expose a capability. And misses a few things you'll need, e.g. the `mount` and `unmount` callbacks. For `ObjectDevice`s, those can be received via [this interface](https://github.com/fnuecke/oc2/blob/1.18-forge/src/main/java/li/cil/oc2/api/bus/device/object/LifecycleAwareDevice.java).
